src/ipns/dns.rs

```diff
@@ -74,9 +76,10 @@ mod tests {
     use crate::tests::async_test;
 
     #[test]
+    #[ignore]
```

Comment:
Forgot this while jumping between commits, was expecting it to be removed: Why was this ignore added?

Reply:
Because domain uses tokio-timer to timeout dns queries, and they require a tokio executor, they timeout immediately, meaning that the test will fail with a timeout error. As a side note I've been working on hashgraph the past week and think that we could remove the entire ipns/dns stuff in favor of a hashgraph based solution in the not too distant future.

Reply:
Oh... Doesn't this mean that the dns side is broken with this PR?

Yeah I was thinking that the whole dependency would need to be switched for trust-dns.

Could you elaborate how hashgraph solution could replace ipns/dns?

Reply:
This PR makes the switch to the async-std executor, but dns was broken before due to the bugfix not being included

Reply:
so with hashgraph I'm working on a byzantine fault tolerant distributed key value store with a dynamic set of peers. If that turns out to work well, it could be ipns 2.0. The other place where I see it being useful in ipfs is in bitswap for addressing the problems I mentioned with it's incentive mechanism. We can discuss it further in a chat if you're interested.

Reply:
Cool! I was just making sure I guessed the scale, as in "ipns 2.0" and "nothing compatible with current ipns" correct. Yeah, let's follow up with this on the chat.